docs/source/testing/mocking-responses.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Mocking responses
+---
+
+Apollo Kotlin provides two tools to mock responses in tests:
+
+- `TestNetworkTransport`: allows to specify the GraphQL responses returned by `ApolloClient`
+- `MockServer`: a low-level HTTP server that can be used to mock any responses
+
+In most cases, `TestNetworkTransport` will be more straightforward to use in your tests, while `MockServer` can be
+useful to test specific server behavior, such as error cases, custom headers, timeouts, etc.
```

Comment:
Maybe something like:

```suggestion
`MockServer` creates a full HTTP server. It is useful to test specific server behaviours, such as error cases, HTTP headers, timeouts, etc. It will also require very little changes to your production code as you'll only need to change `serverUrl`.
`TestNetworkTransport` will replace the HTTP call with a preregistered GraphQL response. It is more lightweight and requires more modifications to your production code.
```

Basically changing `"straighforward"` to `"lightweight"`